android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt

```diff
@@ -106,6 +107,13 @@ open class QuestionnaireActivity : BaseMultiLanguageActivity(), View.OnClickList
           !it.value.isNullOrEmpty() &&
           !it.value.contains(STRING_INTERPOLATION_PREFIX)
       }
+    paramDataActionParams =
+      actionParams.filter {
+        it.paramType == ActionParameterType.PARAMDATA &&
+          !it.value.isNullOrEmpty() &&
+          !it.value.contains(STRING_INTERPOLATION_PREFIX)
+      }
+    questionnaireViewModel.setParamData(paramDataActionParams)
```

Comment:
I think the parameters are pre-loaded and interpolated when the questionnaire is launched. Why do we need to set new params?

Reply:
see the `launchQuestionnaire` function. It accepts a list of params that is provided by the caller.

Reply:
launchQuestionnaire function accepts PREPOPULATED action params only. 
thats why i have initialized separate action params for PARAMDATA type.